src/adaptors/tcp_adaptor.c

```diff
@@ -1727,21 +1727,25 @@ static void qdr_tcp_delivery_update(void *context, qdr_delivery_t *dlv, uint64_t
                DLV_FMT" qdr_tcp_delivery_update: disp: %"PRIu64", settled: %s",
                DLV_ARGS(dlv), disp, settled ? "true" : "false");
 
-        if (settled && disp == PN_RELEASED) {
-            // When the connector is unable to connect to a tcp endpoint it will
-            // release the message. We handle that here by closing the connection.
-            // Half-closed status is signalled by read_eos_seen and is not
-            // sufficient by itself to force a connection closure.
+        const bool final_outcome = qd_delivery_state_is_terminal(disp);
```

Comment:
Out of curiousity, is it possible there has been a change in proton behaviour here?

Reply:
@grs @ssorj - good question.  I've gone back as far as 0.34.0 proton (min required version for skupper-router) and the issue is present: outcome Released, settled=False... hang.

I've added a simple TCP test that verifies the fix in this PR so going forward we'll be sure this fixed for good.